### PR TITLE
Delete umounted disks from node

### PR DESCRIPTION
**Pull request by JacieChao**
- Watch for umounted disks and auto update disk status
- Update node api, UI could delete umounted disks from node

Fixed issue: https://github.com/rancher/longhorn/issues/136

### CODE REVIEW

File: controller/node_controller.go [hunk 1]
```diff
@@ -323,9 +325,21 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
 		}
 		// get disk available size
 		diskInfo, err := nc.getDiskInfoHandler(disk.Path)
-		if err != nil {
-			logrus.Errorf("Get disk information on node %v error: %v", node.Name, err)
+		// if the file system has been changed or the disk has been umount
+		if err != nil || diskInfo == nil || diskInfo.Fsid != diskID {
```
Comment (yasker) [inline controller/node_controller.go]: diskInfo won't be nil if err is nil.

Also, print different logs if err is not nil vs fsid changed. One is cannot detect the disk on the path, the other is filesystem has changed.

File: controller/node_controller.go [hunk 2]
```diff
@@ -323,9 +325,21 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
...
+			logrus.Errorf("File system on node %v has been changed: %v", node.Name, err)
+			// disable invalid disk
+			updateDisk.AllowScheduling = false
+			updateDisk.StorageMaximum = 0
+			updateDisk.StorageReserved = 0
+			diskStatus.State = types.DiskStateUnschedulable
+			diskStatus.StorageAvailable = 0
 		} else {
+			// if user has mount the disk back
```
Comment (yasker) [inline controller/node_controller.go]: Don't need this comment. Just update the StorageMaximum. In fact, you can always update the StorageMaximum.

Also, don't update `AllowScheduling` to true automatically. I'd like to have the user do that manually since he/she manually mounts the disk back.

File: controller/node_controller.go [hunk 3]
```diff
@@ -323,9 +325,21 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
...
-		if err != nil {
-			logrus.Errorf("Get disk information on node %v error: %v", node.Name, err)
+		// if the file system has been changed or the disk has been umount
+		if err != nil || diskInfo == nil || diskInfo.Fsid != diskID {
+			logrus.Errorf("File system on node %v has been changed: %v", node.Name, err)
+			// disable invalid disk
+			updateDisk.AllowScheduling = false
+			updateDisk.StorageMaximum = 0
+			updateDisk.StorageReserved = 0
+			diskStatus.State = types.DiskStateUnschedulable
```
Comment (yasker) [inline controller/node_controller.go]: Don't need to set it to unschedulable manually here. The calculation should cover it.

File: manager/node.go [hunk 4]
```diff
@@ -69,15 +69,28 @@ func (m *VolumeManager) DiskUpdate(name string, updateDisks []types.DiskSpec) (*
...
 				logrus.Warnf("StorageMaximum has been changed for disk %v of node %v. Detected maximum storage %v, current setting %v", diskInfo.Path, name, diskInfo.StorageMaximum, uDisk.StorageMaximum)
 			}
+			diskUpdateMap[diskInfo.Fsid] = uDisk
 		} else {
-			// add disks
-			if uDisk.StorageMaximum != 0 && uDisk.StorageMaximum != diskInfo.StorageMaximum {
-				logrus.Warnf("StorageMaximum has been changed for disk %v of node %v. Detected maximum storage %v, current setting %v", diskInfo.Path, name, diskInfo.StorageMaximum, uDisk.StorageMaximum)
-			} else {
-				uDisk.StorageMaximum = diskInfo.StorageMaximum
+			// check whether disk has been changed file system
```
Comment (yasker) [inline manager/node.go]: `file system has changed on the disk`.

Same below

**Comment by yasker** [comment]
The commit titles are too generic. Be short and descriptive.

File: controller/node_controller.go [hunk 5]
```diff
@@ -323,21 +325,37 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
...
+			diskStatus.StorageAvailable = 0
+		} else if diskInfo == nil || diskInfo.Fsid != diskID {
+			logrus.Errorf("disk %v on node %v has changed file system", disk.Path, node.Name)
+			// disable invalid disk
+			updateDisk.AllowScheduling = false
+			updateDisk.StorageMaximum = 0
+			updateDisk.StorageReserved = 0
+			diskStatus.StorageAvailable = 0
 		} else {
+			if updateDisk.StorageMaximum == 0 {
```
Comment (JacieChao) [inline controller/node_controller.go]: To deal with change `storageMaximum` for tests, I still remain this option. Otherwise disks storageMaximum will always change to real size and couldn't test edge case.
